src/Compilers/Core/MSBuildTaskTests/TargetTests.cs

```diff
@@ -390,6 +390,115 @@ public void GenerateEditorConfigCoreHandlesMalformedCompilerVisibleItemMetadata(
+        [InlineData(".NETStandard", "1.5", "7.3")]
+        [InlineData(".NETStandard", "2.0", "7.3")]
+        [InlineData(".NETStandard", "2.1", "8.0")]
+
+        [InlineData("UnknownTFM", "0.0", "7.3")]
+        public void LanguageVersionGivenTargetFramework(string tfi, string tfv, string expectedVersion)
+        {
+            XmlReader xmlReader = XmlReader.Create(new StringReader($@"
+<Project>
+    <PropertyGroup>
+        <TargetFrameworkIdentifier>{tfi}</TargetFrameworkIdentifier>
+        <_TargetFrameworkVersionWithoutV>{tfv}</_TargetFrameworkVersionWithoutV>
+    </PropertyGroup>
+    <Import Project=""Microsoft.CSharp.Core.targets"" />
+</Project>
+"));
+
+            var instance = CreateProjectInstance(xmlReader);
+            instance.Build(GetTestLoggers());
+
+            var langVersion = instance.GetPropertyValue("LangVersion");
+            var maxLangVersion = instance.GetPropertyValue("MaxSupportedLangVersion");
+
+            Assert.Equal(expectedVersion, langVersion);
+            Assert.Equal(expectedVersion, maxLangVersion);
+        }
+
+        [Fact]
+        public void ExplicitLangVersion()
+        {
+            XmlReader xmlReader = XmlReader.Create(new StringReader($@"
+<Project>
+    <PropertyGroup>
+        <TargetFrameworkIdentifier>.NETCoreApp</TargetFrameworkIdentifier>
+        <_TargetFrameworkVersionWithoutV>2.0</_TargetFrameworkVersionWithoutV>
+        <LangVersion>55.0</LangVersion>
+    </PropertyGroup>
+    <Import Project=""Microsoft.CSharp.Core.targets"" />
+</Project>
+"));
+
+            var instance = CreateProjectInstance(xmlReader);
+            instance.Build(GetTestLoggers());
+
+            var langVersion = instance.GetPropertyValue("LangVersion");
+            var maxLangVersion = instance.GetPropertyValue("MaxSupportedLangVersion");
+
+            Assert.Equal("55.0", langVersion);
+            Assert.Equal("7.3", maxLangVersion);
+        }
+
+        [Fact]
+        public void ExplicitMaxLanguageVersion()
+        {
+            XmlReader xmlReader = XmlReader.Create(new StringReader($@"
+<Project>
+    <PropertyGroup>
+        <MaxSupportedLangVersion>55.0</MaxSupportedLangVersion>
+    </PropertyGroup>
+    <Import Project=""Microsoft.CSharp.Core.targets"" />
+</Project>
+"));
+
+            var instance = CreateProjectInstance(xmlReader);
+            instance.Build(GetTestLoggers());
+
+            var langVersion = instance.GetPropertyValue("LangVersion");
+            var maxLangVersion = instance.GetPropertyValue("MaxSupportedLangVersion");
+
+            Assert.Equal("55.0", langVersion);
+            Assert.Equal("55.0", maxLangVersion);
+        }
+
+        [Fact]
+        public void ExplicitLanguageVersionOverridesExplicitMaxLanguageVersion()
+        {
+            XmlReader xmlReader = XmlReader.Create(new StringReader($@"
+<Project>
+    <PropertyGroup>
+        <MaxSupportedLangVersion>55.0</MaxSupportedLangVersion>
```

Comment:
Would remove this test because we don't, or at least shouldn't, support the explicit setting of `MaxSUpportedLangVersion`.